src/parser.rs

```diff
@@ -17,17 +17,17 @@ pub fn is_empty_c_type(c_type: &str) -> bool {
 }
 
 impl Library {
-    pub fn read_file(&mut self, dir: &Path, lib: &str) -> Result<(), String> {
-        let file_name = make_file_name(dir, lib);
+    pub fn read_file(&mut self, dir: &Path, libs: &mut Vec<String>) -> Result<(), String> {
+        let file_name = make_file_name(dir, &libs[libs.len() - 1]);
```

Comment:
Maybe add `assert!(!libs.is_empty())`?` before this?

Reply:
Or check and return error.

Reply:
It cannot be empty since the call stack starts with `read_file` with one element, the only other place where we push/pop is the read_repository function. We add one element, call `read_file` then once it has returned, it pops what it just added. The array always has at least one element.